buildutils/src/ensure-package.ts

```diff
@@ -138,17 +168,20 @@ export async function ensurePackage(
 
   // Template the CSS index file.
   if (cssImports && fs.existsSync(path.join(pkgPath, 'style/base.css'))) {
-    let cssIndex = CSS_HEADER.trim();
+    const funcName = 'ensurePackage';
+    let cssIndexContents = utils.fromTemplate(
+      HEADER_TEMPLATE,
+      { funcName },
+      { end: '' }
+    );
     cssImports.forEach(cssImport => {
-      cssIndex += `\n@import url('~${cssImport}');`;
+      cssIndexContents += `\n@import url('~${cssImport}');`;
     });
-    cssIndex += "\n\n@import url('./base.css');\n";
-    const cssPath = path.join(pkgPath, 'style/index.css');
-    const prev = fs.readFileSync(cssPath, { encoding: 'utf8' });
-    if (prev !== cssIndex) {
-      messages.push(`Updated ./${data.style}`);
-      fs.writeFileSync(cssPath, cssIndex);
-    }
+    cssIndexContents += "\n\n@import url('./base.css');\n";
+
+    // write out cssIndexContents, if needed
+    const cssIndexPath = path.join(pkgPath, 'style/index.css');
+    messages.push(...ensureFile(cssIndexPath, cssIndexContents));
```

Comment:
I think the previous code would create the file if it didn't exist, but this function will fail?

Reply:
If the file doesn't exist, the old code will raise an error when it tries to read at:

```typescript
const prev = fs.readFileSync(cssPath, { encoding: 'utf8' });
```

The new code will instead print a warning if the file doesn't exist. Not creating the file makes sense, since this would limit possible weird consequences if the `buildutils` is out of sync with the rest of the packages, for whatever reason.

Reply:
If the file doesn't exist, the old code would raise an error when it tried to read:

```typescript
const prev = fs.readFileSync(cssPath, { encoding: 'utf8' });
```

The new code will print a warning. Not creating the file makes sense, since it minimizes possible consequences if the version of `buildutils` is out of sync with the rest of the packages.

Reply:
> If the file doesn't exist, the old code would raise an error when it tried to read

Ok, I agree that is good to clean up. Another issue though: Does not the packagess with only an `index.css` (and no `base.css`) need this treatment? Or is the way SVGs are consumed intrinsically link them to the base/index split setup?

Reply:
This section of the code has nothing to do with the new svg handling. I needed to copy some functionality from this section and use it several times in `ensureUiComponents`, so I encapsulated that functionality in a new `ensureFile` function. I then cleaned up the old code by replacing it with `ensureFile`, which also has the benefit of giving it the more graceful error handling.

> Or does the way SVGs are consumed intrinsically link them to the base/index split setup?

I don't think there's any cases in which a split `index.css` and `base.css` are actually required. We just split them up that way so that one part (`index.css`) can be 100% automatically managed by `buildutils`.